dl/src/main/scala/com/intel/analytics/bigdl/nn/Reshape.scala

```diff
@@ -50,15 +50,6 @@ class Reshape[@specialized(Float, Double) T: ClassTag](
   }
 
   override def updateOutput(input: Tensor[T]): Tensor[T] = {
-
-    if ((batchMode.nonEmpty && !batchMode.get) ||
-          (input.nElement() == nElement && batchMode.isEmpty && input.size(1) != 1)) {
```

Comment:
@psyyz10  mind taking a look at this? Maybe batchMode can be either true or false instead of Option[Boolean]